packages/priority-queue/src/main/PriorityQueue.ts

```diff
@@ -47,11 +50,16 @@ export class PriorityQueue {
       queueObject.priority = priority;
       queueObject.reject = reject;
       queueObject.resolve = resolve;
-      queueObject.retry = Number(this.config.maxRetries) >= 0 ? Number(this.config.maxRetries) : queueObject.retry;
+      queueObject.retry = 0;
```

Comment:
Instead of counting down it's counting upwards which is better for the growth calculation